applications/RANSModellingApplication/custom_strategies/generic_convergence_criteria_mpi.h

```diff
@@ -0,0 +1,227 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya (https://github.com/sunethwarna)
+//
+
+#ifndef KRATOS_GENERIC_CONVERGENCE_CRITERIA_MPI_H
+#define KRATOS_GENERIC_CONVERGENCE_CRITERIA_MPI_H
+
+/* Project includes */
+#include "includes/define.h"
+#include "includes/model_part.h"
+#include "solving_strategies/convergencecriterias/convergence_criteria.h"
+#include "utilities/openmp_utils.h"
+
+namespace Kratos
+{
+///@addtogroup IncompressibleFluidApplication
+///@{
+
+///@name Kratos Classes
+///@{
+
+/// Convergence criteria for fluid problems.
+/**
+ This class implements a convergence control based on nodal velocity and
+ pressure values. The error is evaluated separately for each of them, and
+ relative and absolute tolerances for both must be specified.
+ */
+template <class TSparseSpace, class TDenseSpace>
+class MPIGenericConvergenceCriteria : public ConvergenceCriteria<TSparseSpace, TDenseSpace>
```

Comment:
why do you need an mpi-specific one?

Reply:
Because I have to check whether specific dof is in my partition in the case of MPI, thsi involves and if condition in the inner most loop. Thats the reason why i created the mpi specific one.